scripts/core.js

```diff
@@ -121,18 +145,24 @@ var Substitution = (() => {
    *  Creates a substitution from the given JSON.
    */
   Substitution.fromJSON = function(json) {
-    let inputType = this.prototype.INPUT_TYPE_STRINGS.indexOf(json.inputType);
-    return new Substitution(json.input, json.output, json.caseSensitive, inputType);
+    const inputType = this.prototype.INPUT_TYPE_STRINGS.indexOf(json.inputType);
+    const outputType = this.prototype.OUTPUT_TYPE_STRINGS.indexOf(json.outputType);
```

Comment:
Here you should check if the JSON really has an `outputType`, otherwise it fails. Look how the mode is filled in `SubstitutionGroup.fromJSON` for reference.